exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -191,6 +199,48 @@ func (pq *persistentQueue[T]) restoreQueueSizeFromStorage(ctx context.Context) (
 	return bytesToItemIndex(val)
 }
 
+// marshalCurrentMetadata constructs metadata from current pq state and returns the marshaled bytes.
+// This is a helper and does not perform the Set operation itself.
+// pq.mu must be held.
+// TODO: Enable when https://github.com/open-telemetry/opentelemetry-collector/issues/12890 is done
+//
+//nolint:unused
+func (pq *persistentQueue[T]) marshalCurrentMetadata() ([]byte, error) {
```

Comment:
Consider to remove marshalCurrentMetadata and persistCurrentMetadata from this PR (save the content) and will be in the next PR.